### ISSUE TITLE
webhook engine does not work with tornado 4.5.2

### ISSUE THREAD

**Issue by viq** [saltstack/salt#45590]
### Description of Issue/Question
When JSON request comes in, the response is `500 internal server error`

### Setup
```
/etc/salt/master.d/engines.conf 
engines:
  - webhook:
      port: 8000
      address: 127.0.0.1
```
https://prometheus.io/docs/alerting/configuration/#webhook_config pointed at http://localhost:8000/prometheus/alertmanager

### Steps to Reproduce Issue
Alertmanager sends JSON, but it doesn't seem relevant. Logs include:
```
2018-01-21 21:52:25,664 [tornado.application][ERROR   ][31253] Uncaught exception POST /prometheus/alertmanager (127.0.0.1)
HTTPServerRequest(protocol='http', host='localhost:8000', method='POST', uri='/prometheus/alertmanager', version='HTTP/1.1', remote_ip='127.0.0.1', headers={'Host': 'localhost:8000', 'User-Agent': 'Alertmanager/0.9.1', 'Content-Type': 'application/json', 'Content-Length': '850', 'Accept-Encoding': 'gzip'})
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/tornado/web.py", line 1509, in _execute
    result = method(*self.path_args, **self.path_kwargs)
  File "/usr/lib/python2.7/site-packages/salt/engines/webhook.py", line 74, in post
    fire('salt/engines/hook/' + tag, payload)
  File "/usr/lib/python2.7/site-packages/salt/engines/webhook.py", line 65, in fire
    fire_master(msg, tag)
  File "/usr/lib/python2.7/site-packages/salt/utils/event.py", line 714, in fire_event
    dump_data = self.serial.dumps(data)
  File "/usr/lib/python2.7/site-packages/salt/payload.py", line 181, in dumps
    return msgpack.dumps(msg, use_bin_type=use_bin_type)
  File "/usr/lib/python2.7/site-packages/msgpack/__init__.py", line 47, in packb
    return Packer(**kwargs).pack(o)
  File "msgpack/_packer.pyx", line 274, in msgpack._packer.Packer.pack
  File "msgpack/_packer.pyx", line 276, in msgpack._packer.Packer.pack
  File "msgpack/_packer.pyx", line 224, in msgpack._packer.Packer._pack
  File "msgpack/_packer.pyx", line 271, in msgpack._packer.Packer._pack
TypeError: can't serialize <tornado.httputil.HTTPHeaders object at 0x6005353be490>
2018-01-21 21:52:25,665 [tornado.access   ][ERROR   ][31253] 500 POST /prometheus/alertmanager (127.0.0.1) 1.90ms
```

@gtmanfred says `looks like they changed the HEADERs to be an object instead of a dictionary`

### Versions Report
```
salt-call --versions-report
/usr/lib/python2.7/site-packages/msgpack/__init__.py:47: FutureWarning: use_bin_type option is not specified. Default value of the option will be changed in future version.
  return Packer(**kwargs).pack(o)
Salt Version:
           Salt: 2017.7.2
 
Dependency Versions:
           cffi: 1.11.2
       cherrypy: 13.1.0
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: 2.0.3
      gitpython: 2.1.8
          ioflo: Not Installed
         Jinja2: 2.10
        libgit2: 0.26.0
        libnacl: 1.6.1
       M2Crypto: 0.27.0
           Mako: 1.0.7
   msgpack-pure: Not Installed
 msgpack-python: 0.5.0
   mysql-python: Not Installed
      pycparser: 2.18
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: 0.26.3
         Python: 2.7.14 (default, Sep 20 2017, 01:25:59)
   python-gnupg: 0.3.9.dev0
         PyYAML: 3.12
          PyZMQ: 16.0.3
           RAET: Not Installed
          smmap: 2.0.3
        timelib: Not Installed
        Tornado: 4.5.2
            ZMQ: 4.2.2
 
System Versions:
           dist:   
         locale: UTF-8
        machine: x86_64
        release: 4.14.13-1-hardened
         system: Linux
        version: Not Installed
```

**Comment by gtmanfred** [comment]
This should be fixed by #45589

**Comment by viq** [comment]
Indeed that commit fixes this, with the updated engine I was able to receive an event.
